sensor/kubernetes/listener/resources/virtualmachine/dispatcher/virtualmachines.go

```diff
@@ -51,6 +51,10 @@ func processVirtualMachine(vm *virtualmachine.Info, action central.ResourceActio
 	} else {
 		store.AddOrUpdate(vm)
 	}
+	vmState := virtualMachineV1.VirtualMachine_STOPPED
+	if vm.Running {
+		vmState = virtualMachineV1.VirtualMachine_RUNNING
+	}
```

Comment:
I'm thinking; maybe we need to `Get` the vm from the store to set the VSOCK here too. Otherwise if we get an update from the VM after the instance is processed we will send `0` as VSOCK. That or we should consider this case in the central pipeline. WDYT?

Reply:
I meant VM, instance and scan index data all being sent to a single pipeline.
The first two are indeed already sent by the same channel to central. The latter currently has a dedicated isolated flow (except for the check that the VM exists in the index processing in sensor).

Reply:
I think I'll stick to two central pipelines for now, but will make sure pre-existing state and VSock CID are propagated from the already stored info to the upserted one.